# Configuration file for the feeding demo with the Gen2 arm
#
# If not otherwise specified:
#     All units are metric.
#     All forces and torques are in N and Nm, respectively.
#     All distances are in m.
#     All rotations are in radians.
#     All robot configurations specify 6 joint angles in a list
#     All positiosn are [x, y, z], and quaternions are [w, x, y, z]
#

# Main.cpp Params
watchdogTimeout: 0.05 #50ms until watchdog dies
visualization:
  topicName: dart_markers/feeding
  baseFrameName: map

# Ada Node Params
velocityLimits:
  standard: [0.5, 0.5, 0.6, 0.6, 0.6, 0.6]
  approach: [0.3, 0.3, 0.3, 0.3, 0.3, 0.3]

# FT Sensor Params
ftSensor:
  controllerFTThresholdTopic: /move_until_touch_topic_controller/set_forcetorque_threshold/
  ftTopic: /forque/forqueSensor
  thresholds:
    standard:       # used when no other thresholds are specified
      force: 4
      torque: 4
    afterGrabFood:  # active when going up after food has been grabbed
      force: 50
      torque: 2

# Perception parameters
perception:
  detectorDataUri: package://pr_assets/data/objects/tag_data_foods.json
  referenceFrameName: j2n6s200_link_base
  rateSeconds: 0.5
  timeoutSeconds: 0.5

# Workspace Params
# Pose relative to robot base
workspace:
  wheelchair:
    urdf: package://pr_assets/data/furniture/wheelchair.urdf
    pos: [0.0, 0.0, 0.005]
    quat: [1.0, 0.0, 0.0, 0.0]
  plate:
    urdf: package://pr_assets/data/objects/plate.urdf
    pos: [0.255, -0.29, 0.205]
    quat: [1.0, 0.0, 0.0, 0.0]
  table:
    urdf: package://pr_assets/data/furniture/table_feeding.urdf
    pos: [0.06, -0.48, -0.515]
    quat: [1.0, 0.0, 0.0, 0.0]

# Hard-coded Configurations
configs:
  wheelchair:
    # Ready for Transfer (In Front)
    in_front_person: [-2.30252, 4.23221, 3.84109, -4.65546, 3.94225, 4.26543]
    
    # Ready for Transfer (Side)
    #in_front_person: [2.74709, 2.01400, 1.85257, -2.83523, -1.61925, -2.67325]

    # Ready for Acquisition
    above_plate: [-2.11666, 3.34967, 2.04129, -2.30031, -2.34026, 2.9545]

# Acquistion Action Library
action_selection:
  food_names: ["banana", "cantaloupe", "carrot", "grape", "strawberry"]
  action_keys: [5, 1, 1, 3, 1]
  actions:
    -
      pre_pos: [0.0, 0.0, 0.0]
      pre_quat: [1.0, 0.0, 0.0, 0.0]
      pre_offset: [0.0, 0.0, 0.0]
      pre_force: 0.0
      pre_torque: 0.0
      grasp_twist: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
      grasp_duration: 0.0
      grasp_force: 0.0
      grasp_torque: 0.0
      extract_twist: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0]
      extract_duration: 0.0
      extract_force: 0.0
      extract_torque: 0.0

# Acquisition Parameters
move_above:
  tsr_bounds: [0.005, 0.005, 0.005, 0.01, 0.01, 0.01]
  distance: 0.05
move_into:
  overshoot: 0.0 # how far to plan past food
move_out:
  z_max: 0.05

# Transfer Parameters
move_face:
  overshoot: -0.06 # How far from mouth to stop
move_away:
  offset: [0.0, -0.05, 0.0]